### DISCUSSION TITLE
关于打卡位置变更的问题

### DISCUSSION THREAD

**Discussion by lagran** [lxysl/Auto_Attendance#15] [Q&A] [answered]
作者你好，我使用脚本之前打卡地点一直在学校，但最近发现在打卡页面定位后会提示与上次打卡不在同一个城市。奇怪的是在打卡未中断的情况下第一次位置变更是如何提交通过的，还有一直以来打卡地点是在哪里？希望在被辅导员查水表之前能帮我解除这些疑惑，谢谢~
![16092914461411310](https://user-images.githubusercontent.com/27532563/103324443-93c07680-4a82-11eb-914f-47b60de3a5cd.jpeg)

**Comment by lxysl** [comment, marked as answer]
感谢您的反馈，我已经对代码作出调整，增加了打卡信息中的城市信息。如果使用原来的代码也请不必担心，因为打卡定位看的是地理位置信息，这个和城市信息是分开的。新增的城市信息不影响原有的功能，只是为了避免出现当前地点与上次不在同一城市的问题，避免辅导员不必要的麻烦。
关于程序第一次如何定位的问题，其实程序不会自己定位，他只会拉取和复制上一次提交的打卡信息（不管是自动打卡还是手动打卡），只是对其中的时间等作出调整。
如果您要使用最新的代码，可以从本仓库给你的仓库创建一个pull request，然后merge。为了避免麻烦，我会帮您完成创建pull request这一步，请在您的仓库中merge代码，程序会在merge后自动进行一次打卡，并开启后续的定时打卡。如果不清楚操作步骤的话，暴力一点直接删库重新fork也是可以的。如有帮助请点个star支持一下，祝使用愉快！

**Comment by lagran** [comment]
感谢您详细的解答，您的程序很有用，已Starred！
